File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.PersonService.Models/save-person_result.ts


import { PersonLanguage_Item } from './person-language_item';
import { ISavePerson_Result } from './isave-person_result';

export class SavePerson_Result implements ISavePerson_Result {
  
	public PersonID?: number;
	public FirstMiddleNames: string = "";
	public LastNames: string = "";
	public Gender: string = "";
	public UnitID?: number;
	public IsUSCitizen: boolean = false;
	public NationalID: string = "";
	public ResidenceStreetAddress: string = "";
	public ResidenceCityID?: number;
	public ContactEmail: string = "";
	public ContactPhone: string = "";
	public DOB?: Date;
	public POBCityID?: number;
	public FatherName: string = "";
	public MotherName: string = "";
	public HighestEducationID?: number;
	public FamilyIncome?: number;
	public EnglishLanguageProficiencyID?: number;
	public PassportNumber: string = "";
	public PassportExpirationDate?: Date;
	public PassportIssuingCountryID?: number;
	public LawPoliceMilitaryID: string = "";
	public Rank?: number;
	public YearsInCurrentPosition?: number;
	public MedicalClearanceStatus?: boolean;
	public MedicalClearanceDate?: Date;
	public DentalClearanceStatus?: boolean;
	public DentalClearanceDate?: Date;
	public PsychologicalClearanceStatus?: boolean;
	public PsychologicalClearanceDate?: Date;
	public ModifiedByAppUserID?: number;
	public Languages?: PersonLanguage_Item[];
  
	public ErrorMessages: string[] = null;
}

File: GTTS/INL.PersonService.Models/SavePerson_Result.cs
using INL.Services.Models;
using System;
using System.Collections.Generic;

namespace INL.PersonService.Models
{
    public class SavePerson_Result : BaseResult, ISavePerson_Result
    {
        public long? PersonID { get; set; }
        public string FirstMiddleNames { get; set; }
        public string LastNames { get; set; }
        public char Gender { get; set; }
        public long? UnitID { get; set; }
        public bool IsUSCitizen { get; set; }
        public string NationalID { get; set; }
        public string ResidenceStreetAddress { get; set; }
        public int? ResidenceCityID { get; set; }
        public string ContactEmail { get; set; }
        public string ContactPhone { get; set; }
        public DateTime? DOB { get; set; }
        public int? POBCityID { get; set; }
        public string FatherName { get; set; }
        public string MotherName { get; set; }
        public int? HighestEducationID { get; set; }
        public decimal? FamilyIncome { get; set; }
        public int? EnglishLanguageProficiencyID { get; set; }
        public string PassportNumber { get; set; }
        public DateTime? PassportExpirationDate { get; set; }
        public int? PassportIssuingCountryID { get; set; }
        public string LawPoliceMilitaryID { get; set; }
        public int? Rank { get; set; }
        public int? YearsInCurrentPosition { get; set; }
        public bool? MedicalClearanceStatus { get; set; }
        public DateTime? MedicalClearanceDate { get; set; }
        public bool? DentalClearanceStatus { get; set; }
        public DateTime? DentalClearanceDate { get; set; }
        public bool? PsychologicalClearanceStatus { get; set; }
        public DateTime? PsychologicalClearanceDate { get; set; }
        public int? ModifiedByAppUserID { get; set; }
        public List<PersonLanguage_Item> Languages { get; set; }
    }
}
